hal/src/stm32f2xx/rtc_hal.c

```diff
@@ -84,21 +101,187 @@ void HAL_RTC_Initialize_UnixTime()
     /* Get calendar_time date struct values */
     RTC_DateStructure.RTC_WeekDay = 6;
     RTC_DateStructure.RTC_Date = 1;
-    RTC_DateStructure.RTC_Month = 0;
+    RTC_DateStructure.RTC_Month = 1;
     RTC_DateStructure.RTC_Year = 0;
 
     setRTCTime(&RTC_TimeStructure, &RTC_DateStructure);
 }
 
+static void hal_rtc_configure()
+{
+    /* Enable RTC Clock */
+    RCC_RTCCLKCmd(ENABLE);
+
+    /* Wait for RTC registers synchronization */
+    RTC_WaitForSynchro();
+
+    /* RTC register configuration done only once */
+    if (RTC_ReadBackupRegister(RTC_BKP_DR0) != 0xC1C1)
+    {
+        RTC_InitTypeDef RTC_InitStructure;
+        uint32_t AsynchPrediv = 127; // 127 + 1
+#if HAL_PLATFORM_INTERNAL_LOW_SPEED_CLOCK
+        uint32_t SynchPrediv = (RCC->BDCR & 0x00000300) != (HSE_RTC_PRESCALER & 0x300) ? 255 : 8124; // (255 + 1) or (8124 + 1)
+#else
+        uint32_t SynchPrediv = 255; // (255 + 1)
+#endif // HAL_PLATFORM_INTERNAL_LOW_SPEED_CLOCK
+
+        /* Configure the RTC data register and RTC prescaler */
+        RTC_InitStructure.RTC_AsynchPrediv = AsynchPrediv;
+        RTC_InitStructure.RTC_SynchPrediv = SynchPrediv;
+        RTC_InitStructure.RTC_HourFormat = RTC_HourFormat_24;
+
+        /* Check on RTC init */
+        if (RTC_Init(&RTC_InitStructure) != ERROR)
+        {
+            /* Configure RTC Date and Time Registers if not set - Fixes #480, #580 */
+            /* Set date/time to 2000/01/01 00:00:00 */
+            HAL_RTC_Initialize_UnixTime();
+
+            /* Indicator for the RTC configuration */
+            RTC_WriteBackupRegister(RTC_BKP_DR0, 0xC1C1);
+        }
+    }
+}
+
+static void hal_rtc_configure_clock()
```

Comment:
Prefer to return something in case of failure to conditionally execute following procedures.

Reply:
This is handled by asserts.